File: zyx/Leetcode/src/easy/Code977.java
package easy;

import java.util.Arrays;

/**
 * 977: 给你一个按 非递减顺序 排序的整数数组 nums，返回 每个数字的平方 组成的新数组，要求也按 非递减顺序 排序。
 * @author tony
 */
public class Code977 {


    public static int[] sortedSquares(int[] nums) {
        //事实证明，sorted所谓的自然排序就是升序排序
        return Arrays.stream(nums).map(x -> x * x).sorted().toArray();
    }

    public static int[] sortedSquares2(int[] nums) {
         if (nums == null || nums.length < 1) {
             return null;
         }
         int[] ans = new int[nums.length];
         for (int i = 0; i < nums.length; ++i) {
             ans[i] = nums[i] * nums[i];
         }
         Arrays.sort(ans);
         return ans;
    }



    public static void main(String[] args) {
        int[] nums = {-4,-1,0,3,10};
        Arrays.stream(sortedSquares(nums)).forEach(System.out::println);
    }

}

File: wendy/Leetcode/easy/code977.js
/*
题目：给你一个按"非递减顺序"排序的整数数组nums，返回每个数字的平方组成的新数组，要求也按"非递减顺序"排序。
*/
/*
解题思路：使用map函数对数组进行操作，item代表对每个元素都要操作；
         使用sort函数对数组进行排序，参数(a,b)=>a-b代表，a>b时，a在后面，a<b时，b在后面
*/
var sortedSquares = function(nums) {
    return nums.map(item => item * item).sort((a,b) => a - b)
 };